Nova-LMS-Project/test/com/lms/dao/test/AuthorDaoTest.java

```diff
@@ -0,0 +1,80 @@
+package com.lms.dao.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import com.lms.dao.AuthorDaoImpl;
+
+import com.lms.model.Author;
+
+public class AuthorDaoTest {
+	
+	private Author author;
+	private AuthorDaoImpl authorDao;
+	private int newAuthorId;
+	
+	@BeforeEach
+	void init() throws FileNotFoundException, IOException {
+		authorDao = new AuthorDaoImpl();
+		author = new Author(789, "test 1");
+		newAuthorId = authorDao.save(author);
+		
+	}
+	
+	@AfterEach
+	void tearThis() throws FileNotFoundException, IOException {
+		// WARNING maybe something that doesn't call the method we are trying to test
+		authorDao.delete(newAuthorId);
+	}
+
+	@Test
+	public void saveTest() throws FileNotFoundException, IOException {
+		authorDao.delete(newAuthorId);
+		
+		int previousCount;
+		previousCount = authorDao.findAll().size();
+		newAuthorId = authorDao.save(author);
+		int currentCount = authorDao.findAll().size();
+		assertTrue(previousCount < currentCount);
```

Comment:
Throughout, all test assertions should include a message (the optional final String parameter) describing what is being asserted. (Here, e.g., you ware asserting that the number of authors in the database has increased after a new author has been saved, so `assertTrue(previousCount < currentCount, "Saving new author causes row count to increase")`.)